docs/control_server.md

```diff
@@ -0,0 +1,102 @@
+Already connected websockets are not affected by closing the gate.
+
+The RESTful endpoint that controls this is `/api/v2/device/gate`.
+Note that the control_port for the control server is not the same as the port for the websocket server.
+
+* `GET host:control_port/api/v2/device/gate` returns a JSON message indicating the status of the gate.
+For example, `{"open": true, "timestamp": "2009-11-10T23:00:00Z"}` indicates that the gate
+is open and has been open since the given timestamp.  
+Similarly, `"open": false` indicates that the gate is closed.
+* `POST/PUT/PATCH host:control_port/api/v2/device/gate?open=<boolean>` raises or lowers the gate.
+The value of the open parameter may be anything that `golang` will parse as a boolean (see [ParseBool](https://godoc.org/strconv#ParseBool)).
+This endpoint is idempotent.
+Any attempt to open the gate when it is already open or close it when it is already closed results in a **200** status.
+If this endpoint did change the status of the gate, a **201** status is returned.
+
+### Metrics
+
+`xmidt_talaria_gate_status` is the exposed Prometheus metric that indicates the status of the gate.
+When this gauge is 0.0, the gate is closed.  When this gauge is 1.0, the gate is open.
+
+## Connection Drain
+Talaria supports the draining of websocket connections.
+Essentially, this means shedding load in a controlled fashion.
+Websocket connections can be drained at a given rate, e.g. `100 connections/minute`,
+or can be drained as fast as possible.
+
+Only 1 drain job is permitted to be running at any time.
+Attempts to start a drain when one is already active results in an error.
+
+**IMPORTANT**:  The device gate may be open when a drain is started.
+That means that devices can connect and disconnect during a drain.
+In order to prevent a situation where the drain job cannot ever complete,
+computations about the number of devices are done once when the job is started.
+For example, if a drain job is instructed to drain all devices, the count of devices
+is computed at the start of the job and exactly that many devices are drained.
+This may mean that connections are left at the end of a drain when the gate is not closed.
+If this behavior is not desired, *close the device gate before starting a drain.*
+
+The RESTful endpoint that controls the connection drain is `/api/v2/device/drain`.
+Note that the control_port for the control server is not the same as the port for the websocket server.
+
+* `GET host:control_port/api/v2/device/drain` returns a JSON message indicating whether
+a drain job is active and the progress of the active job if one is running.
+If a drain has previously completed, the information about that job will be
+available via this endpoint until a new drain job is started.
+
+* `POST/PUT/PATCH host:control_port/api/v2/device/drain` attempts to start a drain job.
+This endpoint returns the same JSON message as a `GET` when it starts a drain job,
```

Comment:
I thought I already commented here but what does the get return? Information about the new job or information about the previous job?